domains/measurements/Measurements.API/API/Controllers/MeasurementsController.cs

```diff
@@ -0,0 +1,33 @@
+using API.Models;
+using API.Models.Request;
+using API.Services;
+using EnergyOriginAuthorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Controllers;
+
+[ApiController]
+[Authorize]
+public class MeasurementsController : AuthorizationController
+{
+    readonly ILogger<MeasurementsController> logger;
+    readonly IMeasurementsService measurementsService;
+
+    public MeasurementsController(ILogger<MeasurementsController> logger, IMeasurementsService measurementsService)
+    {
+        this.logger = logger;
+        this.measurementsService = measurementsService;
+    }
+
+    /// <summary>
+    /// Get consumption measurements for all metering point connected to current user
+    /// </summary>
+    /// <param name="request"></param>
+    /// <returns></returns>
+    [HttpGet]
+    [Route("measurements/consumption")]
+    public async Task<ConsumptionResponse> GetMeasurements([FromQuery] MeasurementsRequest request)
```

Comment:
@robertrosborg What is the advantage of creating a input class instead of individual parameters?